utils/dbutil/compactdb/compactdb.go

```diff
@@ -0,0 +1,99 @@
+package compactdb
+
+import (
+	"errors"
+	"math/big"
+	"strconv"
+
+	"github.com/Fantom-foundation/lachesis-base/kvdb"
+	"github.com/Fantom-foundation/lachesis-base/kvdb/table"
+
+	"github.com/Fantom-foundation/go-opera/utils"
+)
+
+type contCompacter struct {
+	kvdb.Store
+	prev []byte
+}
+
+func (s *contCompacter) Compact(_ []byte, limit []byte) error {
+	err := s.Store.Compact(s.prev, limit)
+	if err != nil {
+		return err
+	}
+	s.prev = limit
+	return nil
+}
+
+func compact(db *contCompacter, prefix []byte, iters int) error {
+	// use heuristic to locate tables
+	nonEmptyPrefixes := make([]byte, 0, 256)
+	for b := 0; b < 256; b++ {
+		if !isEmptyDB(table.New(db, append(prefix, byte(b)))) {
+			nonEmptyPrefixes = append(nonEmptyPrefixes, byte(b))
+		}
+	}
+	if len(nonEmptyPrefixes) == 0 {
+		return nil
+	}
+	if len(nonEmptyPrefixes) != 1 && len(nonEmptyPrefixes) < 50 {
```

Comment:
If I understand correctly when `len(nonEmptyPrefixes) < 50` we will compact the db similar to `len(nonEmptyPrefixes) == 1` a.k.a at `// once a table is located, split the range into *iters* chunks for compaction` . Can you explain a little bit more about it, please?

Reply:
It splits it into 3 cases:
== 1:
000001
000002
...
000FFF

It's probably a single table. If keys are numbers, then most likely they all will be behind `00` prefix. An example is epochs storage, which would be behind `00` prefix, because first byte is 0, unless there's more epochs than 2^24

> 1, < 50:
It's probably a collection of tables. An example is gossip DB, which has 34 tables

> 50:
It's probably a single table - that's more prefixes than there's tables in Opera. An example is a collection of transactions or MPTs - both would have 256 non-empty prefixes